core/decorators/async-setup-decorator.ts

```diff
@@ -5,6 +5,7 @@ import { SETUP } from "./_metadata-keys";
 export function AsyncSetup(target: object,
                            decoratedPropertyKey: string,
                            descriptor?: TypedPropertyDescriptor<() => any>) {
+   descriptor = undefined; // Unused
```

Comment:
Depends how strict you want to be, I try to be as strict as possible and having the compiler inform you that a parameter isn't being used I think is a good idea. I do which there was a UNUSED macro like I have in C/C++. I also tried to use `<void>descriptor;` and that satisfied the compiler but the runtime didn't like it :)